moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -200,6 +200,15 @@
     // Package-private to control subclasses.
```

Comment:
nit: should we move the other assignments into this constructor like JsonValueReader's `stack` did?